hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileUpdateHandler.java

```diff
@@ -104,7 +120,23 @@ public void handle(@NonNull final HandleContext handleContext) throws HandleExce
         fileStore.put(builder.build());
     }
 
-    private void resolveMutableBuilderAttributes(
+    private static void handleUpdateUpgradeFile(FileUpdateTransactionBody fileUpdate, HandleContext handleContext) {
+        final var fileStore = handleContext.writableStore(WritableUpgradeStore.class);
+        // empty old upgrade file
+        fileStore.resetFileContents();
+        final var file = new File.Builder()
+                .fileId(FileID.newBuilder()
+                        .fileNum(fileUpdate.fileIDOrElse(FileID.DEFAULT).fileNum())
```

Comment:
This can be `fileIDOrThrow`, or `fileServiceConfig.upgradeFileNumber()`. The orElse for default is incorrect (default file has the wrong file number), and also should never happen, as we've already acquired the fileID correctly before calling this method.